datadog/resource_datadog_synthetics_test_.go

```diff
@@ -1399,6 +1400,26 @@ func syntheticsBrowserStepParams() schema.Schema {
 					Type:        schema.TypeString,
 					Optional:    true,
 				},
+				"pattern": {
+					Description: `Pattern to use for an "extractFromEmailBody" step.`,
+					Type:        schema.TypeList,
+					MaxItems:    1,
+					Elem: &schema.Resource{
+						Schema: map[string]*schema.Schema{
+							"value": {
+								Description: "Pattern to use for the step.",
+								Type:        schema.TypeString,
+								Optional:    true,
+							},
+							"type": {
+								Description: "Type of pattern to use for the step.",
```

Comment:
As this is in fact an enum, I think it would be better to add the possible values. They would be auto generated if we had the explicited in the api spec, but we don't...
```suggestion
								Description: "Type of pattern to use for the step.   Valid values are `regex`, `x_path`.",
```